src/v/pandaproxy/client/consumer.cc

```diff
@@ -307,10 +321,114 @@ ss::future<kafka::offset_commit_response> consumer::offset_commit(
     return req_res(std::move(req_builder));
 }
 
-ss::future<shared_consumer_t>
-make_consumer(shared_broker_t coordinator, kafka::group_id group_id) {
+ss::future<kafka::fetch_response>
+consumer::consume(std::chrono::milliseconds timeout, int32_t max_bytes) {
+    using broker_reqs_t
+      = absl::flat_hash_map<shared_broker_t, kafka::fetch_request>;
+
+    // Split requests by broker
+    broker_reqs_t broker_reqs;
+    for (auto const& [t, ps] : _assignment) {
+        for (const auto& p : ps) {
+            auto broker = co_await _brokers.find(model::topic_partition{t, p});
+            auto& session = _fetch_sessions[broker];
+
+            auto& req = broker_reqs
+                          .try_emplace(
+                            broker,
+                            kafka::fetch_request{
+                              .replica_id = model::node_id{-1}, // consumer
+                              .max_wait_time = timeout,
+                              .min_bytes = 0,
+                              .max_bytes = max_bytes,
+                              .isolation_level = 0, // READ_UNCOMMITTED
+                              .session_id = session.id,
+                              .session_epoch = session.epoch,
+                            })
+                          .first->second;
+            auto offset = session.offsets[t]
+                            .try_emplace(p, model::offset{0})
+                            .first->second;
+
+            if (req.topics.empty() || req.topics.back().name != t) {
+                req.topics.push_back(kafka::fetch_request::topic{.name{t}});
+            }
+
+            req.topics.back().partitions.push_back(
+              kafka::fetch_request::partition{
+                .id = p,
+                .fetch_offset = offset,
+                .partition_max_bytes = max_bytes});
+        }
+    }
+
+    // TODO Ben:
+    // * Size & time limit - round-robin
+    co_return co_await ss::map_reduce(
+      std::make_move_iterator(broker_reqs.begin()),
+      std::make_move_iterator(broker_reqs.end()),
+      [this](
+        broker_reqs_t::value_type br) -> ss::future<kafka::fetch_response> {
+          auto b = br.first;
+          auto& req = br.second;
+          ppclog.info("Consumer: {}, fetch_req: {}", *this, req);
+          auto res = co_await b->dispatch(std::move(req));
+          ppclog.info("Consumer: {}, fetch_res: {}", *this, res);
+
+          if (res.error != kafka::error_code::none) {
+              co_return res;
+          }
+
+          auto& session = _fetch_sessions[b];
```

Comment:
a code comment above here, explainig what is the actual logic trying to do would be helpful. There are lot of metadata changes here that is hard for me to keep track of being new to this tree.

Reply:
I think this will move into fetch_session.